DOC: recommend not distributing generated code (#6201)
cython/cython: docs/src/userguide/source_files_and_compilation.rst

Distributing Cython modules
----------------------------

> .. note::
> 
>     This section is historical and contains recommendations that outlived their
>     purpose. Following recent improvements in the distribution toolchain, it is
>     now recommended to *not* include generated files in distributions,
>     and require Cython at build-time as defined in
>     `PEP 518 <https://www.python.org/dev/peps/pep-0518/>`_
>     and `PEP 621 <https://www.python.org/dev/peps/pep-0621/>`_.
>     See :ref:`basic_setup.py`.

Reviewer: I think we can now also make this paragraph a normal part of the text:
```suggestion
Following recent improvements in the distribution toolchain, it is
not recommended to include generated files in source distributions.
Instead, `require` Cython at build-time to generate the C/C++ files,
as defined in `PEP 518 <https://www.python.org/dev/peps/pep-0518/>`_
and `PEP 621 <https://www.python.org/dev/peps/pep-0621/>`_.
See :ref:`basic_setup.py`.
```

> It is possible to distribute the generated ``.c`` files as well
> as your Cython sources, so that users can install your module without needing

Reviewer: ```suggestion
It is, however, possible to distribute the generated ``.c`` files together with
your Cython sources, so that users can install your module without needing
```

Distributing Cython modules
----------------------------

.. note::

    This section is historical and contains recommendations that outlived their
    purpose. Following recent improvements in the distribution toolchain, it is
    now recommended to *not* include generated files in distributions,
    and require Cython at build-time as defined in
    `PEP 518 <https://www.python.org/dev/peps/pep-0518/>`_
    and `PEP 621 <https://www.python.org/dev/peps/pep-0621/>`_.

> It is strongly recommended that you distribute the generated ``.c`` files as well

Reviewer: I think we can keep much of this section, just with a more optional wording.
```suggestion
It is possible to distribute the generated ``.c`` files as well
```
etc.
